src/timepicker/__tests__/time-picker.e2e.js

```diff
@@ -220,4 +221,36 @@ describe('TimePicker', () => {
       expect(option1).toBe('00:15');
     });
   });
+  describe('when using moment', () => {
```

Comment:
This first test was just to establish that the provider was working, and get the ball rolling on the PR.  Long term, it seems like the ideal solution would be to create some kind of setup where all of the tests defined in this file could be rerun for every date-io function type.  This would have required a massive refactoring of this file and would have made it much harder to follow, so I decided to hold off on that, but I do think it would be the most complete way to ensure that the timepicker is fully working with the other support date libraries.

Reply:
Agreed, this looks like a good first step